src/base/net/downloadhandler.cpp

```diff
@@ -147,25 +147,22 @@ void Net::DownloadHandler::checkDownloadSize(qint64 bytesReceived, qint64 bytesT
 
 bool Net::DownloadHandler::saveToFile(const QByteArray &replyData, QString &filePath)
 {
-    QTemporaryFile *tmpfile = new QTemporaryFile(Utils::Fs::tempPath() + "XXXXXX");
-    if (!tmpfile->open()) {
-        delete tmpfile;
+    QTemporaryFile tmpfile {Utils::Fs::tempPath() + "XXXXXX"};
+    tmpfile.setAutoRemove(false);
+
+    if (!tmpfile.open())
         return false;
-    }
 
-    tmpfile->setAutoRemove(false);
-    filePath = tmpfile->fileName();
+    filePath = tmpfile.fileName();
     qDebug("Temporary filename is: %s", qUtf8Printable(filePath));
     if (m_reply->isOpen() || m_reply->open(QIODevice::ReadOnly)) {
-        tmpfile->write(replyData);
-        tmpfile->close();
+        tmpfile.write(replyData);
+        tmpfile.close();
         // XXX: tmpfile needs to be deleted on Windows before using the file
         // or it will complain that the file is used by another process.
-        delete tmpfile;
         return true;
     }
     else {
-        delete tmpfile;
         Utils::Fs::forceRemove(filePath);
```

Comment:
This might fail while the object isn't destroyed because the file will be open.

Reply:
You're right, fixed.